core/src/main/scala/zio/prelude/ops/package.scala

```diff
@@ -0,0 +1,40 @@
+package zio.prelude
+
+import zio.prelude.fx.ZPure
+
+package object ops {
+
+  implicit class EnrichZPure(z: ZPure.type) {
+
+    def whenCase[W, S1, R, E, A, B](a: => A)(
+      pf: PartialFunction[A, ZPure[W, S1, S1, R, E, B]]
+    ): ZPure[W, S1, S1, R, E, Option[B]] =
+      ZPure.suspend(pf.lift(a).map(_.map(Some(_))).getOrElse(ZPure.succeed(None)))
+
+    def unless[W, S1, R, E, A](condition: => Boolean)(
+      pure: => ZPure[W, S1, S1, R, E, A]
+    ): ZPure[W, S1, S1, R, E, Option[A]] =
+      when[W, S1, R, E, A](!condition)(pure)
+
+    def when[W, S1, R, E, A](condition: => Boolean)(
+      pure: => ZPure[W, S1, S1, R, E, A]
+    ): ZPure[W, S1, S1, R, E, Option[A]] =
+      ZPure.suspend(if (condition) pure.map(Some(_)) else ZPure.succeed(None))
+
+    def forEachDiscard[W, S1, R, E, A](as: Iterable[A])(
```

Comment:
Oh they didn't exist?

Reply:
There’s `forEach_` which works on things with an implicit `Foreach`, but the inference seemed somewhat troublesome (and for maps I had to convert them to lists first)

Reply:
These operators are built in now.

Reply:
Removed